app/components/certificate/description.tsx

```diff
@@ -38,6 +48,20 @@ export default function DescriptionSection({
                 <Text fontWeight="bold">Expires On</Text>
                 <Text>{validToFormatted}</Text>
               </VStack>
+
+              <Flex justifyContent="flex-end">
+                <Form method="post" onSubmit={() => setIsDisabled(true)}>
```

Comment:
Could you use `useTransition`? and check if it is in the submitting state?

https://remix.run/docs/en/1.15.0/hooks/use-transition#transitionstate

Reply:
It should redirect to pending when you click submit so I don't think it's necessary